File: src/main/java/com/nao4j/subtrol/service/CostService.java
package com.nao4j.subtrol.service;

import com.nao4j.subtrol.document.internal.Costs;
import com.nao4j.subtrol.document.internal.ExactPeriod;
import org.springframework.security.access.annotation.Secured;

import static com.nao4j.subtrol.security.Roles.ADMIN;
import static com.nao4j.subtrol.security.Roles.USER;

public interface CostService {

    @Secured({USER, ADMIN})
    Costs calculateForPeriod(String userId, ExactPeriod period);

}

File: src/main/kotlin/com/nao4j/subtrol/service/CostService.kt
package com.nao4j.subtrol.service

import com.nao4j.subtrol.model.Costs
import com.nao4j.subtrol.model.ExactPeriod
import com.nao4j.subtrol.security.ADMIN
import com.nao4j.subtrol.security.USER
import org.springframework.security.access.annotation.Secured

interface CostService {

    @Secured(value = [USER, ADMIN])
    fun calculateForPeriod(userId: String, period: ExactPeriod): Costs

}
